data_management/story_photo_color_transformations.py

```diff
@@ -0,0 +1,155 @@
+        self.invert_output.pack()
+        self.invert_output["command"] = self.invert_check_box
+        self.save_btn = tk.Button(self.controls_frame)
+        self.save_btn["text"] = "save"
+        self.save_btn["command"] = self.save_button
+        self.save_btn.pack(side="top")
+
+        self.quit = tk.Button(self.controls_frame, text="QUIT", fg="red", command=self.master.destroy)
+        self.quit.pack(side="bottom")
+
+        self.next_phase_btn = tk.Button(self.controls_frame)
+        self.next_phase_btn["text"] = "Next Phase"
+        self.next_phase_btn["command"] = self.next_phase_button
+        self.next_phase_btn.pack(side="right")
+
+        self.image_handle = None
+
+    def next_phase_button(self):
+        """
+        Sets a flag that helps in advancing to the next phase
+        :return: None
+        """
+        self.goto_next_phase_flag = True
+        self.master.destroy()
+
+    def invert_check_box(self):
+        """
+        Checking then unchecking this box results in an image with text and background colors inverted
+        """
+        self.invert_output = not self.invert_output
+        self.update_image('red', self.red_slider.current_sorted_values)
+
+    def update_image(self, channel, values):
+        """
+        This function updates the image based on user input from slider selections
+        """
+
+        def normalize(C):
+            C = C - C.min()
+            C = C / C.max()
+            C = C * 255
+            return C
+
+        if channel == 'red':
+            position = 0
+
+        if channel == 'green':
+            position = 1
+
+        if channel == 'blue':
+            position = 2
+
+        r1, r2 = self.red_slider.current_sorted_values
+        g1, g2 = self.green_slider.current_sorted_values
+        b1, b2 = self.blue_slider.current_sorted_values
+
+        print(r1, r2, g1, g2, b1, b2)
+
+        selectionR = (self.np_img_orig[:, :, position] >= r1) * (self.np_img_orig[:, :, position] <= r2)
+        selectionG = (self.np_img_orig[:, :, position] >= g1) * (self.np_img_orig[:, :, position] <= g2)
+        selectionB = (self.np_img_orig[:, :, position] >= b1) * (self.np_img_orig[:, :, position] <= b2)
+
+        if self.invert_red == True:
+            selectionR = np.invert(selectionR)
+        selection = selectionR * selectionG * selectionB
+
+        selection = np.reshape(selection, (self.np_img.shape[0], self.np_img.shape[1], 1)).astype('uint8')
+
+        if self.invert_output == True:
+            self.np_img = normalize(255 - np.multiply(self.np_img_orig, selection, dtype="uint8"))
+        else:
+            self.np_img = np.multiply(self.np_img_orig, selection, dtype="uint8")
+
+        self.redraw()
+
+
+if __name__ == "__main__":
+    root = tk.Tk()
+    # Resize the display window
+    root.geometry("800x1000")
+    app = Application(master=root, next_phase=None)
+    app.mainloop()
```

Comment:
Very impressive code blocks here. Hopefully I can use some fo it for snippet preprocessing. Either way, Good job to you and your Team!